### PR TITLE
feat: support expire prometheus metrics

### LINKED ISSUE #9627: bug:  apisix 内存管理-过期监控key不释放问题
### Current Behavior

如果打开了prometheus插件，并且upstream使用了k8s服务发现或者upstream ip随着发布而改变的话，在apisix中就会产生过多的监控key，从而导致内存不断增长，如果不重启apisix最终OOM

### Expected Behavior

我期望在upstream ip改变的时候有一个自动检测机制，把内存里的监控指标中不存在的node节点的key进行释放

### Error Logs

_No response_

### Steps to Reproduce

我通过将指标中的node纬度关闭从而规避了因发布而导致的upstream ip 改变产生过多key的问题
image

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by monkeyDluffy6017**
### Description

The prometheus metrics we record are stored in LRUs, and although there is no memory leakage, the metrics are not eliminated until they reach the upper limit of the LRUs. As the metrics become more and more numerous, the prometheus server consumes more and more CPU when pulling data, and a mechanism is needed to eliminate the outdated data.

Fixes #9627

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by ShenFeng312** [comment]
LGTM!

File: conf/config-default.yaml [hunk 1]
```diff
@@ -584,22 +584,23 @@ plugin_attr:          # Plugin attributes
...
+    #   bandwidth:
+    #     extra_labels:
+    #       - upstream_addr: $upstream_addr
     # default_buckets:
     #   - 10
     #   - 50
     #   - 100
     #   - 200
     #   - 500
+    # expire: 86400                         # the expiration time after metrics become inactive, unit: second
```
Comment (moonming) [inline conf/config-default.yaml:603]: Default value should be 0 to keep the same strategy as before, no expiration time

**Comment by theweakgod** [comment]
👍

File: conf/config-default.yaml (hunk 1 above)
Comment (monkeyDluffy6017) [reply conf/config-default.yaml]: Done

**Review by shreemaan-abhishek — DISMISSED** [review]
LGTM

File: conf/config-default.yaml [hunk 2]
```diff
@@ -600,7 +600,9 @@ plugin_attr:          # Plugin attributes
     #   - 100
     #   - 200
     #   - 500
-    # expire: 86400                         # the expiration time after metrics become inactive, unit: second
+    # expire: 0                       # The expiration time after metrics become inactive, unit: second.
+                                      # 0 means the metrics will not expire
+                                      # We recommend using 600 If you need to set the expiration time
```
Comment (membphis) [inline conf/config-default.yaml:605]: should avoid to use `we`

How about this:

If you need to set the expiration time, it is recommended to use 600, which is 10 minutes.

Comment (monkeyDluffy6017) [reply conf/config-default.yaml]: Done

**Review by membphis — APPROVED** [review]
LGTM

**Review by AlinsRan — APPROVED** [review]

**Comment by god12311** [comment]
@monkeyDluffy6017 Hello, if I want to fix this issue in version 3.2.2 of APISIX, can I make the changes according to the code in this link (https://github.com/apache/apisix/pull/10869/files), or can I directly replace the prometheus/exporter.lua file? Looking forward to your reply.

**Pull request merged**